internal/kubeclient/event.go

```diff
@@ -0,0 +1,139 @@
+package kubeclient
```

Comment:
This package has been driving me a little crazy for a while. I think we've had a tendency to dump a lot of unrelated k8s things in here.

Right now I see indexers, rountrippers, helpers for patching resources, error-handling stuff.

I might prefer that we put this code somewhere else, but I don't have any specific concerns about the implementation -- just the location.

Reply:
There are already lots of packages in the `internal` package now, I think it would be better to group `kubernetes` related resources into one. So created a new package named `kubernetes` and moved event related packages under `kubernetes/event` in https://github.com/akuity/kargo/pull/1859/commits/a8e1c2853f624cc100d62d4a414828e0f1106708 :)